File: NoteS/models/entity/SyntaxType.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;

namespace NoteS.models.entity;

[Table("custom_types_syntax")]
public class SyntaxType
{
    [Column("id")] [Key] public int Num { get; init; }
    [Column("syntax")] public string Name { get; init; }
    public static readonly SyntaxType Plain = new(0, "PLAINTEXT");
    public static readonly SyntaxType Markdown = new(1, "MARKDOWN");

    public static int? TypeToNum(SyntaxType? syntaxType)
    {
        return syntaxType?.Num;
    }

    [return: NotNullIfNotNull(nameof(num))]
    public static SyntaxType? NumToType(int? num)
    {
        if (num == null) return null;
        switch (num)
        {
            case 0: return Plain;
            case 1: return Markdown;
        }

        throw new ArgumentException(); //TODO: другая ошибка
    }

    protected SyntaxType()
    {
    }

    protected SyntaxType(int num, string name)
    {
        Num = num;
        Name = name;
    }
}

File: javaBack/ResourceServer/src/main/java/com/notes/models/entity/SyntaxType.java
package com.notes.models.entity;

import lombok.Getter;

@Getter
public enum SyntaxType {
   PLAINTEXT(0),
   MARKDOWN(1);
   final int id;

   SyntaxType(int id) {
      this.id = id;
   }

   public static SyntaxType valueOf(int i) {
      return switch (i) {
         case 0 -> PLAINTEXT;
         case 1 -> MARKDOWN;
         default -> throw new IllegalArgumentException("Unknown syntax type: " + i);
      };
   }
}
